src/main/java/sk/hfa/blog/web/BlogController.java

```diff
@@ -9,52 +10,50 @@
 
-    public BlogController(IImageService imageService, IBlogService blogService) {
-        this.imageService = imageService;
+    public BlogController(IBlogService blogService) {
         this.blogService = blogService;
     }
 
     @PreAuthorize("hasRole('ADMIN')")
-    @PostMapping(produces = MediaType.APPLICATION_JSON_VALUE, consumes = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<MessageResource> createBlogArticle(@RequestBody BlogArticleRequest request) {
-        String operation = "was successfully created.";
-        String message = "Creating a new blog article.";
-
-        if (!Objects.isNull(request.getBlogArticle().getId())) {
-            imageService.deleteBlogArticleImage(request.getBlogArticle().getId());
-            message = "Updating an existing blog article.";
-            operation = "was successfully updated.";
-        }
+    @PostMapping(produces = MediaType.APPLICATION_JSON_VALUE, consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+    public ResponseEntity<MessageResource> createBlogArticle(@Valid BlogArticleDto request) throws JsonProcessingException {
+        log.info("Creating a new blog article.");
+        BlogArticle blogArticle = blogService.save(request);
+        MessageResource responseBody = new BlogArticleMessageResource(blogArticle);
+        log.info("The blog article with the ID: [" + blogArticle.getId() + "] was successfully created.");
+        return ResponseEntity.ok(responseBody);
+    }
 
-        log.info(message);
-        BlogArticle blogArticle = blogService.save(BlogArticle.build(request.getBlogArticle()));
-        MessageResource responseBody = new BlogArticleMessageResource(BlogArticleDto.build(blogArticle, false));
-        log.info("The blog article with the ID: [" + blogArticle.getId() + "] " + operation);
+    @PreAuthorize("hasRole('ADMIN')")
+    @PutMapping(produces = MediaType.APPLICATION_JSON_VALUE, consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+    public ResponseEntity<MessageResource> updateBlogArticle(@Valid  BlogArticleDto request) throws JsonProcessingException {
+        log.info("Updating an existing blog article.");
+        BlogArticle blogArticle = blogService.save(request);
+        MessageResource responseBody = new BlogArticleMessageResource(blogArticle);
+        log.info("The blog article with the ID: [" + blogArticle.getId() + "] was successfully updated.");
         return ResponseEntity.ok(responseBody);
     }
 
     @GetMapping(path = "/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
     public ResponseEntity<MessageResource> getBlogArticle(@PathVariable long id) {
         log.info("Fetching the blog article with the ID: " + id);
         BlogArticle blogArticle = blogService.findById(id);
-        MessageResource responseBody = new BlogArticleMessageResource(BlogArticleDto.build(blogArticle, false));
+        blogArticle.getContent().clear();
+        MessageResource responseBody = new BlogArticleMessageResource(blogArticle);
```

Comment:
Viem ze sme to tak nedodrziavali ani predtym ale mali by sme zacat objekty davat ako final

Reply:
na toto je Kotlin super